### PR TITLE
Repair firebase imports

### PR DESCRIPTION
**Pull request by shans**
As an aside: tslint has control issues.

I mean, I get it, banning 'any' is a good idea. But here? I'm explicitly asking to use any because *it's better that way*. Sure, I could do something like
```
const firebaseInitializeApp = ((firebase as {})["default"] as {})["initializeApp"];
```

except that (1) this isn't fundamentally any safer and (2) it's *way* harder to see what the code is doing. 

The point of any, I thought, was that it was an escape hatch for when you needed to deal with things that were outside the ken of typescript. So why are my only options no-any (ever) or no-dynamic-any (which wouldn't let me do this either)? Where's no-implicit-any, to make sure that I only declare something any when I actually need to?

@mykmartin noticed a similar thing with variable declaration. You can:
* force everyone to always declare variables on one line, always
* force everyone to always declare each variable on its own line, always
* force everyone to always declare adjacent variables on one line, always

in other words, you get lots of *fundamentalist* options. But you can't force variables to be declared on their own line only when they have an initializing value, or any other nuanced combination. So we turned the option off altogether, because we don't like having tslint dictate stupid rules for no other reason than that it can.

I don't think we can turn off no-any without opening ourselves up to a world of hurt, but I'm also pretty convinced that tslint annotations on lines where we need to use any aren't going to go down well in the long run. Is this the point where we fork tslint?

### CODE REVIEW

**Comment by shans** [comment]
Ah! the compiler has a noImplicitAny option; so we *can* disable no-any in tslint. Thankfully.

Paul, any objection to disabling it now? Or would you rather wait until we can turn on noImplicitAny?

**Comment by lindner** [comment]
So I'm not a js module expert, but I was following some hints.  Firebase specifies this syntax for typescript:

```
import firebase from 'firebase';
var app = firebase.initializeApp({ ... });
```

But the first line is not working in our environment.  It fails with 

```
TS1192: Module '".../arcs/main/node_modules/firebase/index"' has no default export.
```

Looking into TypeScript issues I found these two mentions of using the alternate syntax:

```
 import * as firebase from 'firebase';
```

See the following discussions:
- https://github.com/Microsoft/TypeScript/issues/3337
- https://github.com/Microsoft/TypeScript/issues/2242#issuecomment-83694181

I'd hope we can find some solution that avoids using 'default' above.  Will look into this a little bit more.

**Comment by sjmiles** [comment]
I hope I'm not adding noise to this situation, but I have my own fingers in this pie. :)

Firebase still refuses to be loadable from the client without a transpilation step, so I gave up on that, but I still would like a single dependency to avoid, e.g.
```
// keep in sync with shell/source/ArcsLib.js
```

So what I did was create `runtime/firebase.js` as follows:
```
import * as firebase from 'firebase/app';
import 'firebase/database';
import 'firebase/storage';
export {firebase};
```

And then employ that in `firebase-storage.ts` like so:
```
declare var firebase: typeof import('firebase');
import {firebase} from '../../firebase.js';
```

and in `ArcsLib.js` as:
```
import {firebase} from '../../runtime/firebase.js';
```

This way there is only one file to do the firebase incantations (+ one extra decl when using ts).

What say you?

**Comment by shans** [comment]
I've no objection to you hunting around to find a better solution, but in the meantime ToT is broken. Please approve this patch so I can continue to make progress towards our milestone?

I'll also reiterate that this is known brokenness in the modular import approach wrt firebase and typescript: https://github.com/firebase/firebase-js-sdk/issues/716.

**Comment by lindner** [comment]
Another option.  Switch to:

```
import firebase from 'firebase';
```

and add


```
"allowSyntheticDefaultImports": true
```

to `tsconfig.json`

Less hacky, but does return the firebase console message...

**Comment by shans** [comment]
@lindner as I already described on chat last Saturday, the existing tests that demonstrate breakage can be invoked using `./tools/sigh test --manual`.

If you're making further changes in this area be sure to check that these tests continue to pass.
